core/experiments/assets/js/admin/module.js

```diff
@@ -32,6 +39,119 @@ export default class ExperimentsModule extends elementorModules.ViewModule {
 		this.elements.$experimentForm.find( '#submit' ).trigger( 'click' );
 	}
 
+	onExperimentStateChange( e ) {
+		const { experimentId } = e.currentTarget.dataset,
+			experimentNewState = this.getExperimentActualState( experimentId );
+
+		switch ( experimentNewState ) {
+			case STATE_ACTIVE:
+				this.showDependenciesDialog( experimentId );
+				break;
+
+			case STATE_INACTIVE:
+				this.deactivateDependantExperiments( experimentId );
+				break;
+
+			default:
+				break;
+		}
+	}
+
+	getExperimentSelect( experimentId ) {
+		return this.elements.$experimentSelects.filter( `[data-experiment-id="${ experimentId }"]` );
+	}
+
+	setExperimentState( experimentId, state ) {
+		this.getExperimentSelect( experimentId ).val( state );
+	}
+
+	getExperimentActualState( experimentId ) {
+		const state = this.getExperimentSelect( experimentId ).val();
+
+		if ( state !== STATE_DEFAULT ) {
+			return state;
+		}
+
+		// Normalize the "default" state to the actual state value.
+		return this.isExperimentActiveByDefault( experimentId )
+			? STATE_ACTIVE
+			: STATE_INACTIVE;
+	}
+
+	isExperimentActive( experimentId ) {
+		return ( this.getExperimentActualState( experimentId ) === STATE_ACTIVE );
+	}
+
+	isExperimentActiveByDefault( experimentId ) {
+		return ( elementorAdmin.config.experiments[ experimentId ].default === STATE_ACTIVE );
+	}
+
+	areAllDependenciesActive( dependencies ) {
+		return dependencies.every( ( dependency ) => this.isExperimentActive( dependency.name ) );
+	}
+
+	deactivateDependantExperiments( experimentId ) {
+		Object
+			.entries( elementorAdmin.config.experiments )
+			.forEach( ( [ id, experimentData ] ) => {
+				const isDependant = ( experimentData.dependencies.includes( experimentId ) );
+
+				if ( isDependant ) {
+					this.setExperimentState( id, STATE_INACTIVE );
+				}
+			} );
+	}
+
+	showDependenciesDialog( experimentId ) {
+		const experiment = elementorAdmin.config.experiments[ experimentId ];
+
+		const dependencies = experiment
+			.dependencies
+			.map( ( dependencyId ) => (
+				elementorAdmin.config.experiments[ dependencyId ]
+			) );
+
+		if ( this.areAllDependenciesActive( dependencies ) ) {
```

Comment:
You can create some kind of `collection` of the experiments. `ExperimentsCollection`  then
```js
const deps = ExperimentCollection.create( dependencies );

deps.isAllActive()
deps.joinByTitle(', ', ' & ')
deps.activateAll()
deps.deactivateAll()
```